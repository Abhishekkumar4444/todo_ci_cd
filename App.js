// App.js
import React from 'react';
import {SafeAreaView, StyleSheet} from 'react-native';
import {Provider} from 'react-redux';
import TodoList from './src/components/TodoList';
import store from './src/redux/store';

const App = () => {
  return (
    <Provider store={store}>
      <SafeAreaView style={styles.container}>
        <TodoList />
      </SafeAreaView>
    </Provider>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 20,
  },
});

export default App;
