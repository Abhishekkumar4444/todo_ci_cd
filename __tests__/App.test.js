import {render} from '@testing-library/react-native';
import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import App from '../App'; // Adjust the path as necessary

const mockStore = configureStore([]);

describe('App Component Snapshots', () => {
  it('should matches the initial empty todos state snapshot', () => {
    const store = mockStore({
      todos: {
        todos: [], // Initial empty state
        loading: false,
        error: null,
      },
    });

    const {toJSON} = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should matches the snapshot with todos', () => {
    const store = mockStore({
      todos: {
        todos: [
          {id: 1, title: 'Test Todo 1', completed: false},
          {id: 2, title: 'Test Todo 2', completed: true},
        ], // Sample todos
        loading: false,
        error: null,
      },
    });

    const {toJSON} = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should matches the loading state snapshot', () => {
    const store = mockStore({
      todos: {
        todos: [], // Empty todos
        loading: true, // Simulating loading state
        error: null,
      },
    });

    const {toJSON} = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should matches the error state snapshot', () => {
    const store = mockStore({
      todos: {
        todos: [], // Empty todos
        loading: false,
        error: 'Failed to fetch todos', // Simulating error state
      },
    });

    const {toJSON} = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
