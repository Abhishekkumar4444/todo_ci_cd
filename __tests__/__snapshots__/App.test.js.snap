// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Snapshots should matches the error state snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "padding": 20,
      }
    }
    testID="todo"
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Todo"
      style={
        {
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "padding": 8,
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "green",
              },
            ]
          }
        >
          Add Todo
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "top": 300,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 28,
            "fontWeight": "700",
          }
        }
      >
        Loading.....
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`App Component Snapshots should matches the initial empty todos state snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "padding": 20,
      }
    }
    testID="todo"
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Todo"
      style={
        {
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "padding": 8,
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "green",
              },
            ]
          }
        >
          Add Todo
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "top": 300,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 28,
            "fontWeight": "700",
          }
        }
      >
        Loading.....
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`App Component Snapshots should matches the loading state snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "padding": 20,
      }
    }
    testID="todo"
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Todo"
      style={
        {
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "padding": 8,
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "green",
              },
            ]
          }
        >
          Add Todo
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "top": 300,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 28,
            "fontWeight": "700",
          }
        }
      >
        Loading.....
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`App Component Snapshots should matches the snapshot with todos 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "padding": 20,
      }
    }
    testID="todo"
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter Todo"
      style={
        {
          "borderBottomWidth": 1,
          "marginBottom": 10,
          "padding": 8,
        }
      }
      value=""
    />
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              {
                "color": "green",
              },
            ]
          }
        >
          Add Todo
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "justifyContent": "center",
          "top": 300,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 28,
            "fontWeight": "700",
          }
        }
      >
        Loading.....
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
